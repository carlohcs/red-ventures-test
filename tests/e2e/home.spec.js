/**
 * Get a page and
 *
 * @returns {Promise}
 */
async function getHomePage() {
  const page = global.page

  await page.goto(global.config.baseUrl)
  await page.waitForSelector('.page--home')
  console.log(page)
  return page
}

/**
 * Get a page and evaluate a function
 *
 * @param   {Function} fn
 * @returns {Promise}
 */
async function getHomePageAndEvaluate(fn) {
  const page = global.page

  await page.goto(global.config.baseUrl)
  await page.waitForSelector('.page--home')
  return await page.evaluate(fn)
}

jest.setTimeout(global.TEST.TIMEOUT)

describe('Home', () => {
  beforeAll(async () => global.TEST.beforeAll(), global.TEST.TIMEOUT)
  afterAll(async () => global.TEST.afterAll(), global.TEST.TIMEOUT)

  it('Should render home', async () => {
    expect.assertions(1)
    const html = await getHomePageAndEvaluate(
      () => document.querySelector('.page--home').innerHTML
    )

    expect(html).toMatchSnapshot()
  })

  it('Should render server valid results when user change select', async () => {
    const page = await global.page

    // Select the valid options
    await page.select('select[data-step="sun"]', 'low')
    await page.select('select[data-step="water"]', 'rarely')
    await page.select('select[data-step="pets"]', 'true')

    // Wait for server response
    await page.waitForSelector('.page__results__found--active')

    // Get items
    const results = await page.$$('.result__item')

    expect(results.length).toBeGreaterThan(0)
  })

  it('Should render not found results when user change select', async () => {
    const page = await global.page

    // Select the valid options
    await page.select('select[data-step="sun"]')

    // Wait a time for server response
    await page.waitForTimeout(2000)

    // Get non existent items
    const isNotFoundElementActive = await getHomePageAndEvaluate(() =>
      document
        .querySelector('.page__results__not-found')
        .classList.contains('page__results__not-found--active')
    )

    expect(isNotFoundElementActive).toBeFalsy()
  })
})
