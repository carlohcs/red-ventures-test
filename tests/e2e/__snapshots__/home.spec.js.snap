// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should render home 1`] = `
"
    <div class=\\"app\\">
      <!-- HEADER -->
      <section class=\\"page__header\\" id=\\"page__header\\">
        <header class=\\"page__header__logo\\">
          <img src=\\"/logo-white.579818af.svg\\" alt=\\"Greenthumb\\">
        </header>

        <div class=\\"page__header__wrapper\\">
          <div class=\\"page__header__content\\">
            <h1 class=\\"page__header__title\\">Find your next green friend</h1>
            <a href=\\"#page__filters\\">
              <span class=\\"icon icon--arrow-down\\"></span>
            </a>
          </div>
        </div>
      </section>
      <!-- // HEADER -->

      <main class=\\"page__content\\">
        <!-- FILTERS -->
        <section class=\\"page__filters\\" id=\\"page__filters\\">
          <div class=\\"page__filters__wrapper\\">
            <!-- STEPS -->
            <div class=\\"steps\\">
              <div class=\\"steps__wrapper\\">
                <!-- SUNLIGHT -->
                <div class=\\"step\\">
                  <img class=\\"step__icon img-responsive\\" src=\\"/sun.d849b03f.png\\">
                  <div class=\\"step__description\\">
                    <strong>1.</strong> Set the amount of
                    <strong>sunlight</strong> your plant will get.
                  </div>
                  <div class=\\"step__select-wrapper\\">
                    <select class=\\"step__select\\" data-step=\\"sun\\">
                      <option value=\\"\\">Select...</option>
                      <option value=\\"no\\">No sunlight</option>
                      <option value=\\"low\\">Low sunlight</option>
                      <option value=\\"high\\">High sunlight</option>
                    </select>
                  </div>
                </div>
                <!-- // SUNLIGHT -->

                <!-- WATERING CAN -->
                <div class=\\"step\\">
                  <img class=\\"step__icon img-responsive\\" src=\\"/wateringcan.5297a80f.png\\">
                  <div class=\\"step__description\\">
                    <strong>2.</strong> How often do you want to
                    <strong>water</strong> your plant?
                  </div>
                  <div class=\\"step__select-wrapper\\">
                    <select class=\\"step__select\\" data-step=\\"water\\">
                      <option value=\\"\\">Select...</option>
                      <option value=\\"rarely\\">Rarely</option>
                      <option value=\\"regularly\\">Regularly</option>
                      <option value=\\"daily\\">Daily</option>
                    </select>
                  </div>
                </div>
                <!-- // WATERING CAN -->

                <!-- DOG -->
                <div class=\\"step\\">
                  <img class=\\"step__icon img-responsive\\" src=\\"/dog.1398c37e.png\\">
                  <div class=\\"step__description\\">
                    <strong>3.</strong> Do you have pets? Do they
                    <strong>chew</strong> plants?
                  </div>
                  <div class=\\"step__select-wrapper\\">
                    <select class=\\"step__select\\" data-step=\\"pets\\">
                      <option value=\\"\\">Select...</option>
                      <option value=\\"false\\">No/They don't care</option>
                      <option value=\\"true\\">Yes</option>
                    </select>
                  </div>
                </div>
                <!-- // DOG -->
              </div>
            </div>
            <!-- // STEPS -->
          </div>
        </section>
        <!-- // FILTERS -->

        <!-- RESULTS -->
        <section class=\\"page__results\\">
          <!-- NOT FOUND -->
          <div class=\\"page__results__not-found\\">
            <div class=\\"page__results__wrapper\\">
              <div class=\\"page__results__content\\">
                <div class=\\"page__results__content__column\\">
                  <h3 class=\\"page__results__title\\">No results yet...</h3>
                  <p class=\\"page__results__description\\">
                    Use the filters above to find the plant that best fits your
                    environment :)
                  </p>
                </div>
                <div class=\\"page__results__content__column\\">
                  <div class=\\"page__results__illustration\\">
                    <img src=\\"/no-results.bb2f715a.png\\" class=\\"img-responsive\\" alt=\\"Not found!\\">
                  </div>
                </div>
              </div>
            </div>
          </div>
          <!-- // NOT FOUND -->

          <!-- FOUND -->
          <div class=\\"page__results__found\\">
            <div class=\\"page__results__wrapper\\">
              <div class=\\"page__results__illustration\\">
                <img src=\\"/pick.cdf64227.png\\" class=\\"img-responsive\\" alt=\\"Found!\\">
              </div>
              <h3 class=\\"page__results__title\\">Our picks for youh</h3>
            </div>

            <div class=\\"results\\">
              <div class=\\"result__wrapper\\">
                <div class=\\"result__items\\"></div>
              </div>
            </div>
          </div>
          <!-- // FOUND -->
        </section>
        <!-- // RESULTS -->

        <div class=\\"page__back-to-top\\">
          <div class=\\"page__back-to-top__wrapper\\">
            <a class=\\"button button--ghost\\" href=\\"#page__header\\">
              <span class=\\"icon icon--xs icon--arrow-up\\"></span>
              back to the top
            </a>
          </div>
        </div>
      </main>
    </div>

    <!-- SCRIPTS -->
    <script src=\\"/application.f1658735.js\\"></script>
    <!-- // SCRIPTS -->
  

"
`;
